src/doughnut/signingMethods/0/index.js

```diff
@@ -5,9 +5,23 @@ const {
   schnorrkelSign,
   schnorrkelVerify,
 } = require("@polkadot/util-crypto");
+const { isReady } = require('@polkadot/wasm-crypto');
+
+function wait_ms(ms) {
+  var start = Date.now();
+  while (Date.now() - start < ms) {}
+}
+
+// Wrap schnorrkel signing to await wasm crypto readiness
+function sign(payload, keypair) {
+    for(i = 0; i < 10; i++) {
+      if(!isReady()) wait_ms(10);
```

Comment:
blocking-style wait sounds a bad idea.

Reply:
I was hoping you could "hide" as async function but in JS but you can't